Dockerfile.cuda-all

```
@@ -1,49 +1,65 @@
 # syntax=docker/dockerfile:1
 
-FROM nvidia/cuda:12.4.1-cudnn-devel-ubuntu22.04 AS builder
+FROM docker.io/nvidia/cuda:12.9.1-cudnn-devel-ubuntu22.04 AS builder
 
 ARG DEBIAN_FRONTEND=noninteractive
 RUN <<HEREDOC
     apt-get update
     apt-get install -y --no-install-recommends \
         curl \
         libssl-dev \
-        pkg-config
+        pkg-config && \
 
     rm -rf /var/lib/apt/lists/*
 HEREDOC
 
-RUN curl https://sh.rustup.rs -sSf | bash -s -- -y
+RUN curl https://sh.rustup.rs -sSf | bash -s -- -y && test -f /root/.cargo/bin/rustup
 ENV PATH="/root/.cargo/bin:${PATH}"
 RUN rustup update nightly
 RUN rustup default nightly
 
+# MKL build dependencies
+RUN curl -s https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB \
+    | gpg --dearmor | tee /usr/share/keyrings/oneapi-archive-keyring.gpg > /dev/null \
+    && echo "deb [signed-by=/usr/share/keyrings/oneapi-archive-keyring.gpg] https://apt.repos.intel.com/oneapi all main" | \
+    tee /etc/apt/sources.list.d/oneAPI.list \
+    && apt-get update \
+    && apt-get install -y libomp-dev intel-oneapi-mkl-devel
+
 WORKDIR /mistralrs
 
 COPY . .
 
 # Rayon threads are limited to minimize memory requirements in CI, avoiding OOM
 # Rust threads are increased with a nightly feature for faster compilation (single-threaded by default)
-ARG CUDA_COMPUTE_CAP=80
+ARG CUDA_COMPUTE_CAP=70
 ARG RAYON_NUM_THREADS=4
 ARG RUST_NUM_THREADS=4
 ARG RUSTFLAGS="-Z threads=${RUST_NUM_THREADS}"
-ARG WITH_FEATURES="cuda,cudnn"
+ARG WITH_FEATURES="cuda,cudnn,nccl,mkl"
 RUN cargo build --release --workspace --exclude mistralrs-pyo3 --features "${WITH_FEATURES}"
 
-FROM nvidia/cuda:12.4.1-cudnn-runtime-ubuntu22.04 AS base
+FROM docker.io/nvidia/cuda:12.9.1-cudnn-runtime-ubuntu22.04 AS base
 ENV HUGGINGFACE_HUB_CACHE=/data \
     PORT=80
 
 ARG DEBIAN_FRONTEND=noninteractive
+
+RUN curl -s https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB \
+    | gpg --dearmor | tee /usr/share/keyrings/oneapi-archive-keyring.gpg > /dev/null \
+    && echo "deb [signed-by=/usr/share/keyrings/oneapi-archive-keyring.gpg] https://apt.repos.intel.com/oneapi all main" | \
+    tee /etc/apt/sources.list.d/oneAPI.list
+
 RUN <<HEREDOC
     apt-get update
     apt-get install -y --no-install-recommends \
         libomp-dev \
         ca-certificates \
         libssl-dev \
         curl \
-        pkg-config
+        pkg-config \
+        openmpi-bin \
+        intel-oneapi-hpc-toolkit && \
```

Why were the contextual comments removed in https://github.com/EricLBuehler/mistral.rs/pull/1468/commits/444ddc01f0625e6be5d080bedf5cc993f3eab9f0 ?

```suggestion
        # Provided for convenience when using the NCCL crate feature:
        openmpi-bin \
        # Provided for MKL feature runtime:
        intel-oneapi-hpc-toolkit
```

---

They break the `&&`